website_sale_custom_filter/controllers/main.py

```diff
@@ -0,0 +1,133 @@
+                else:
+                    min_max_filter_vals[f_id]["max"] = f_max_val
+        return min_max_filter_vals
+
+    def get_custom_checkbox_filter_values(self):
+        req = dict(request.httprequest.args)
+        value_filter_data = {}
+        if req.get("cust_filter"):
+            filter_data = req["cust_filter"].split("&")
+            for data in filter_data:
+                f_id, val = data.split("-")
+                if f_id in value_filter_data.keys():
+                    value_filter_data[f_id].append(int(val))
+                else:
+                    value_filter_data[f_id] = [int(val)]
+        return value_filter_data
+
+    def get_custom_range_filter_values(self):
+        cust_filter_min_max_vals = self.get_min_max_custom_filters()
+        curr_website = request.env["website"].get_current_website()
+        numerical_filters_available_on_website = request.env[
+            "website.sale.custom.filter"
+        ].search(
+            [("website_ids", "=", curr_website.id), ("filter_type", "=", "numerical")]
+        )
+        range_filter_data = {}
+        # get min max for numerical filters
+        # how to handle if field from filter is related to another model?
+        for num_filter in numerical_filters_available_on_website:
+            col_name = num_filter.numerical_filter_field_id.name
+            model_name = num_filter.numerical_filter_field_id.model
+            if (
+                model_name == "product.product"
+            ):  # there are fields related to product.template sometimes
+                model_name = "product.template"
+            query = f"""
+            SELECT MIN({col_name}), MAX({col_name}) FROM {model_name.replace('.','_')}"""
+            request.env.cr.execute(query)
+            available_min_value, available_max_value = request.env.cr.fetchone()
+
+            curr_filter_id = str(num_filter.id)
+            range_filter_data[curr_filter_id] = {}
+            range_filter_data[curr_filter_id][
+                "available_min_value"
+            ] = available_min_value
+            range_filter_data[curr_filter_id][
+                "available_max_value"
+            ] = available_max_value
+            range_filter_data[curr_filter_id]["min_value"] = available_min_value
+            range_filter_data[curr_filter_id]["max_value"] = available_max_value
+            if cust_filter_min_max_vals and cust_filter_min_max_vals.get(
+                curr_filter_id
+            ):
+                if cust_filter_min_max_vals[curr_filter_id].get("min"):
+                    range_filter_data[curr_filter_id][
+                        "min_value"
+                    ] = cust_filter_min_max_vals[curr_filter_id]["min"]
+                if cust_filter_min_max_vals[curr_filter_id].get("max"):
+                    range_filter_data[curr_filter_id][
+                        "max_value"
+                    ] = cust_filter_min_max_vals[curr_filter_id]["max"]
```

Comment:
Couldn't we split all of this into subfunctions? Something like:
```
def get_custom_range_filter_values(self):
    (...)
    for num_filter in numerical_filters_available_on_website:
        curr_filter_id = str(num_filter.id)
        range_filter_data[curr_filter_id] = {}
        range_filter_data[curr_filter_id].update(num_filter._prepare_filter_default_website_values())
        range_filter_data[curr_filter_id].update(self._prepare_filter_actual_website_values(cust_filter_min_max_vals.get(curr_filter_id)))
    return range_filter_data

def _prepare_filter_actual_website_values(self, min_max_values)
    values = {}
    if min_max_values.get("min"):
        values["min_value"] = min_max_values["min"]
    if min_max_values.get("max"):
        values["max_value"] = min_max_values["max"]
    return values
```
And then on `website.sale.custom.filter` you could have:
```
def _prepare_filter_default_website_values(self):
    self.ensure_one()
    values = {}
    col_name = self.numerical_filter_field_id.name
    model_name = num_filter.numerical_filter_field_id.model
    if (
        model_name == "product.product"
    ):  # there are fields related to product.template sometimes
        model_name = "product.template"
    query = f"""
    SELECT MIN({col_name}), MAX({col_name}) FROM {self.env[model_name]._table}"""
    self.env.cr.execute(query)
    available_min_value, available_max_value = self.env.cr.fetchone()

    values["available_min_value"] = available_min_value
    values["available_max_value"] = available_max_value
    values["min_value"] = available_min_value
    values["max_value"] = available_max_value
    return values
```
I feel it makes the code much more readable with shorter variable names and dedicated functions.